mu-plugins/blocks/global-header-footer/js/view.js

```diff
@@ -193,44 +193,36 @@
 			closeSearchButton.setAttribute( 'aria-label', labels.closeSearchLabel || 'Close Search' );
 		}
 
-		// Watch for the `has-modal-open` class to be removed, and remove the global class too.
-		// This works as a callback to be fired when the global header modals are closed, as
-		// they're attached when each modal opens.
-		const modalCloseObserver = new window.MutationObserver( ( mutationList, observer ) => {
-			for ( const mutation of mutationList ) {
-				const { attributeName, type, target } = mutation;
-				if ( type === 'attributes' && attributeName === 'class' ) {
-					if ( ! target.classList.contains( 'has-modal-open' ) ) {
-						target.classList.remove( 'has-global-modal-open' );
-					}
-				}
-			}
-			// Remove the observer to prevent recursion. This will be re-attached when the modal is opened.
-			observer.disconnect();
-		} );
-
-		const openButtons = document.querySelectorAll( '[data-micromodal-trigger]' );
+		const openButtons = document.querySelectorAll(
+			'.global-header__navigation [data-wp-on-async--click="actions.openMenuOnClick"],' +
```

Comment:
Do we need to rely on the Interactivity API attributes? Is it fair to assume a button in the menu is a dropdown?

Reply:
This is a way to differentiate open and close buttons, if we went for just "button" it would catch both.